pylint/checkers/refactoring/recommendation_checker.py

```diff
@@ -78,38 +78,58 @@ def _check_use_maxsplit_arg(self, node: astroid.Call) -> None:
         """Add message when accessing first or last elements of a str.split() or str.rsplit()."""
 
         # Check if call is split() or rsplit()
-        if (
+        if not (
             isinstance(node.func, astroid.Attribute)
             and node.func.attrname in ("split", "rsplit")
             and isinstance(utils.safe_infer(node.func), astroid.BoundMethod)
         ):
-            try:
-                utils.get_argument_from_call(node, 0, "sep")
-            except utils.NoSuchArgumentError:
-                return
+            return
+
+        try:
+            utils.get_argument_from_call(node, 0, "sep")
+        except utils.NoSuchArgumentError:
+            return
+
+        try:
+            # Ignore if maxsplit arg has been set
+            utils.get_argument_from_call(node, 1, "maxsplit")
+            return
+        except utils.NoSuchArgumentError:
+            pass
 
+        if isinstance(node.parent, astroid.Subscript):
             try:
-                # Ignore if maxsplit arg has been set
-                utils.get_argument_from_call(node, 1, "maxsplit")
+                subscript_value = utils.get_subscript_const_value(node.parent).value
+            except utils.InferredTypeError:
                 return
-            except utils.NoSuchArgumentError:
-                pass
-
-            if isinstance(node.parent, astroid.Subscript):
-                try:
-                    subscript_value = utils.get_subscript_const_value(node.parent).value
-                except utils.InferredTypeError:
-                    return
 
-                if subscript_value in (-1, 0):
-                    fn_name = node.func.attrname
-                    new_fn = "rsplit" if subscript_value == -1 else "split"
-                    new_name = (
-                        node.func.as_string().rsplit(fn_name, maxsplit=1)[0]
-                        + new_fn
-                        + f"({node.args[0].as_string()}, maxsplit=1)[{subscript_value}]"
-                    )
-                    self.add_message("use-maxsplit-arg", node=node, args=(new_name,))
+            # Check for cases where variable (Name) subscripts may be mutated within a loop
+            if isinstance(node.parent.slice, astroid.Name):
+                # Check if loop present within the scope of the node
+                scope = node.scope()
+                for loop_node in scope.nodes_of_class((astroid.For, astroid.While)):
```

Comment:
Could you add type hints for `loop_node` and `assignment_node` below (with `typing.cast`).

I should probably add types to `NodeNG.nodes_of_class` soon, so we don't need to anymore.